### PR TITLE
Supports Array ActivityStreams type 

### LINKED ISSUE #4633: ActivityPub の actor.type が配列である場合に処理を正しく遂行できない
actor.type は配列になり得るが、現状の実装はそれを想定した作りになっていない。というかそもそも `Person` 以外の actor が通らない。（`Service` 一部含め）

https://github.com/syuilo/misskey/blob/7325d66c52365104b6b5d6343324a258470ad2a8/src/remote/activitypub/type.ts#L69

FYI: https://www.w3.org/TR/activitystreams-core/#actors

### LINKED ISSUE #7535: Can't process ActivityStreams objects where the type is an array
## 💡 Summary

[ActivityStreams spec says that `type` may have multiple values](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-type), i.e. it can be an array. However, I discovered that Misskey ignores documents where `type` is an array.

## 🙂 Expected Behavior

Misskey should look through the array of types, pick the most specific one that it supports, and process the document as if it was of that type. For example, if the document has `"type": ["Note", "http://purl.org/goodrelations/v1#Location"]`, and Misskey only supports `Note`, it should process the document as if it said `"type": "Note"`.

## ☹️ Actual Behavior

Misskey ignores documents where the `type` field is an array.

## 📝 Steps to Reproduce

1. Suppose you have an ActivityPub actor somewhere (I just wrote a bunch of static files and put them onto a new sub-domain of my site, aptest.debiania.in.ua);
2. Make this actor post the following activity at https://aptest.debiania.in.ua/notes/2:

    ```json
    {
      "id": "https://aptest.debiania.in.ua/notes/2",
      "type": ["Note", "gr:Location"],
      "published": "2021-05-23T09:06:18.141413+00:00",
      "attributedTo": "https://aptest.debiania.in.ua/actor",
      "content": "Hello, multi-typed world!",
      "to": [
        "https://www.w3.org/ns/activitystreams#Public"
      ],
      "cc": [],
      "inReplyTo": "",
      "summary": "",
      "tag": [],
      "sensitive": false,
      "gr:category": "restaurants/french_restaurants",
      "@context": [
          "https://www.w3.org/ns/activitystreams",
          {
              "gr": "http://purl.org/goodrelations/v1#"
          }
      ]
    }
    ```

    (This is a variation of [an example from the AS-Core spec](https://www.w3.org/TR/activitystreams-core/#fig-an-object-that-is-both-a-place-and-a-gr-location).)
3. Type https://aptest.debiania.in.ua/notes/2 into the search box in Misskey. Observe the logs of your server — you'll notice that Misskey fetches the document, but its UI never shows the post.
4. Optional: try searching for https://aptest.debiania.in.ua/notes/1 This is a simpler document with `"type": "Note"`. It is fetched and displayed just fine, proving that the issue is not with my webserver setup or anything like that.

## 📌 Environment

Here's all the files for my actor: [www.tar.gz](https://github.com/misskey-dev/misskey/files/6531813/www.tar.gz) I simply dropped them into a directory that I served with Nginx. Nginx was configured to set the correct `Content-Type` header on these files:

```
location /notes {
    default_type application/activity+json;
}

location /actor {
    default_type application/activity+json;
}

location /.well-known/webfinger {
    default_type application/jrd+json;
}
```

### PR DESCRIPTION
**Pull request by mei23**
## Summary
Fix #4633
Fix #7535

### CODE REVIEW

File: src/remote/activitypub/models/person.ts [hunk 1]
```diff
@@ -476,7 +476,7 @@ export async function updateFeatured(userId: User['id']) {
 	// Resolve and regist Notes
 	const limit = promiseLimit<Note | null>(2);
 	const featuredNotes = await Promise.all(items
-		.filter(item => item.type === 'Note')
+		.filter(item => getApType(item) === 'Service')
```
Comment (mei23) [inline src/remote/activitypub/models/person.ts]: まちがい

**Comment by syuilo** [comment]
https://github.com/misskey-dev/misskey/pull/7536/files#diff-faa6a9005284933e59001f78ef113a806b179b51c54e514b908b1c58365e5324R6 (IObjectのtype)
の型も変えた方がいいかも？

**Comment by mei23** [comment]
ここね
https://github.com/misskey-dev/misskey/blob/6e8bee7049c0c218e87a674c51d6240b0891fb76/src/remote/activitypub/type.ts#L4-L6

**Comment by syuilo** [comment]
🙏

File: src/remote/activitypub/type.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ export type ApObject = IObject | string | (IObject | string)[];
 
 export interface IObject {
 	'@context': string | obj | obj[];
-	type: string;
+	type: string | unknown[];
```
Comment (mei23) [inline src/remote/activitypub/type.ts:6]: ここって`string |string[]`じゃダメ？

Comment (syuilo) [reply src/remote/activitypub/type.ts:6]: yosasou

**Pull request merged**